docs/specifications/tests/Consistency-TP/consistency01.md

```diff
@@ -1,52 +1,97 @@
 record for the designated zone.   
 
-If the serial number (explained in 3.3.13. of [RFC 1035](https://tools.ietf.org/html/rfc1035)) 
-, which is part of the SOA record, is not consistent between authoritative servers, 
+If the serial number (explained in 3.3.13. of [RFC 1035]), 
+which is part of the SOA record, is not consistent between authoritative servers, 
 there is a possibility that the data served is inconsistent. The reasons for this 
 inconsistency may be different - such as misconfiguration, or as a result of slow 
 propagation to the secondary name servers.
 
 The objective of this test is to verify that the serial number is consistent
 between different authoritative name servers.
 
-For reference purposes : [RFC1982](https://tools.ietf.org/html/rfc1982) 
+For reference purposes : [RFC 1982]
 explains the serial number arithmetic, and section 4.3.5 of 
-[RFC 1034](https://tools.ietf.org/html/rfc1035) explains the importance of
+[RFC 1034] explains the importance of
 serial number consistency.
 
-### Inputs
+## Inputs
 
-The domain name to be tested.
+* The domain name to be tested ("Child Zone").
+* Accepted SOA serial difference ("Accepted Serial Difference"). Default value
+  is 0.
 
-### Ordered description of steps to be taken to execute the test case
+## Ordered description of steps to be taken to execute the test case
 
- 1. Obtain the list of name servers from [Method4] and [Method5].
- 2. Retrieve the SOA RR from all the name servers.
- 3. If the SOA serial number is not the same from all the answers received
-    from step 2, then:
-     1. Check if the difference is minor (An integer should be added to
-        define what is the minor value).
-     2. If the difference is greater than the minor value, then the test
-        case fails.
-     3. If the difference is less than or equal to the minor value,
-        then the test is passed.
+ 1. Obtain the list of name server IPs for the *Child Zone* from [Method4] 
+    and [Method5].
 
-### Outcome(s)
+ 2. Create an SOA query for *Child Zone* apex and send it to each name 
+    server IP.
 
-All authoritative name servers must have consistent serial numbers. If the
-test does not find any inconsistency, then the test case passes.
+ 3. If the name server does not respond with a DNS response, emit 
+    *[NO_RESPONSE]* and go to next name server IP.
 
-### Special procedural requirements	
+ 4. If the name server responds with a DNS response but no a SOA record 
+    is included, emit *[NO_RESPONSE_SOA_QUERY]* and go to next name 
+    server IP.
+
+ 5. Retrieve the SOA SERIAL from the responses and go to next name server
+    IP until all name server IPs have been queried.
+
+ 6. If at least one SOA SERIAL has been retrieved and all serial 
+    numbers are identical, emit *[ONE_SOA_SERIAL]*.
+
+ 7. If at least two serial numbers are different:
+    1. Order the serial number values from smallest to largest following
+       the arithmetic for serial number.
```

Comment:
The specified comparison works for two numbers, but the operator is non-transitive so this step is impossible to implement for arbitrary inputs. For example, given SERIAL_BITS=4, each of the SOA serials 0, 5, 10 is greater than one of the other two and lesser than the other. We should have a correct way to handle zones whose SOA serial have gone bonkers.